nomad/structs/structs.go

```diff
@@ -5746,6 +5746,17 @@ func (j *Job) GetScalingPolicies() []*ScalingPolicy {
 	return ret
 }
 
+// UsesDeployments returns a boolean indicating whether the job configuration
+// results in a deployment during scheduling.
+func (j *Job) UsesDeployments() bool {
+	switch j.Type {
+	case JobTypeService:
+		return true
+	default:
+		return false
+	}
```

Comment:
Could this not be simplified?

```suggestion
        return j.Type == JobTypeService
```

Reply:
It could, but I intentionally did this since a goal is to support Deployments for more job types in the future.  Maybe this is premature?  I was hoping the team would discuss so thanks for taking the bait!